controllers/construct/mongodbstatefulset.go

```diff
@@ -234,8 +246,9 @@ func mongodbContainer(version string, volumeMounts []corev1.VolumeMount) contain
 # run post-start hook to handle version changes
 /hooks/version-upgrade
 
-# wait for config to be created by the agent
-while [ ! -f /data/automation-mongod.conf ]; do sleep 3 ; done ; sleep 2 ;
+# wait for config and keyfile to be created by the agent
+ while ! [ -f /data/automation-mongod.conf -a -f /var/lib/mongodb-mms-automation/authentication/keyfile ]; do sleep 3 ; done ; sleep 2 ;
```

Comment:
can we maybe make `/var/lib/mongodb-mms-automation/authentication/keyfile` a constant in case the filepath changes in the future?

I guess we could also do it for the `automation-mongod.conf` while we're here, what do you think?

Reply:
Sure, will do